### DISCUSSION TITLE
Fetch policy in useQuery

### DISCUSSION THREAD

**Discussion by apenab** [TanStack/query#1959] [Q&A]
I'm in a migration from apollo-client to `react-query`. I have noticed that react-query does not have something similar to the 
 [fetchPolicy](https://www.apollographql.com/docs/react/data/queries/#setting-a-fetch-policy) from apollo-client. There is any way to implement this feature or this feature has some bad points?

**Comment by apenab** [comment]
Maybe this feature could be implemented with a combination of the `staleTime` and `cacheTime` properties?

**Comment by TkDodo** [comment]
I think this could be mapped as follows:

`cache-first`: `staleTime: Infinity`

`cache-only`. No real equivalent because RQ will not throw if there is no data in the cache. You could disable the query and out data in the cache manually and again set `staleTime: Infinity`, but i don’t really get the use-case for this. Your queryFn could also always return a failed promise if you want. 

`cache-and-network`: this the react-query default

`network-only` and `no-cache`: i think you’d need to manually remove or reset the query in the onSuccess callback 🤔
